software/database/src/main/java/brooklyn/entity/database/mysql/MySqlClusterImpl.java

```diff
@@ -0,0 +1,445 @@
+                .configure(MySqlNode.TEMPLATE_CONFIGURATION_URL, MASTER_CONFIG_URL);
+    }
+
+    @Override
+    protected EntitySpec<?> getMemberSpec() {
+        Supplier<Integer> serverIdSupplier = getAttribute(SLAVE_NEXT_SERVER_ID);
+
+        EntitySpec<?> spec = super.getMemberSpec();
+        if (spec != null) {
+            return applyDefaults(spec, serverIdSupplier, SLAVE_CONFIG_URL, true);
+        }
+
+        return EntitySpec.create(MySqlNode.class)
+                .displayName("MySql Slave")
+                .configure(MySqlNode.MYSQL_SERVER_ID, serverIdSupplier.get())
+                .configure(MySqlNode.TEMPLATE_CONFIGURATION_URL, SLAVE_CONFIG_URL)
+                // block inheritance, only master should execute the creation script
+                .configure(MySqlNode.CREATION_SCRIPT_URL, (String) null)
+                .configure(MySqlNode.CREATION_SCRIPT_CONTENTS, (String) null);
+    }
+
+    private EntitySpec<?> applyDefaults(EntitySpec<?> spec, Supplier<Integer> serverId, String configUrl, boolean resetCreationScript) {
+        boolean needsServerId = !isKeyConfigured(spec, MySqlNode.MYSQL_SERVER_ID);
+        boolean needsConfigUrl = !isKeyConfigured(spec, MySqlNode.TEMPLATE_CONFIGURATION_URL.getConfigKey());
+        boolean needsCreationScriptUrl = resetCreationScript && !isKeyConfigured(spec, MySqlNode.CREATION_SCRIPT_URL);
+        boolean needsCreationScriptContents = resetCreationScript && !isKeyConfigured(spec, MySqlNode.CREATION_SCRIPT_CONTENTS);
+        if (needsServerId || needsConfigUrl || needsCreationScriptUrl || needsCreationScriptContents) {
+            EntitySpec<?> clonedSpec = EntitySpec.create(spec);
+            if (needsServerId) {
+                clonedSpec.configure(MySqlNode.MYSQL_SERVER_ID, serverId.get());
+            }
+            if (needsConfigUrl) {
+                clonedSpec.configure(MySqlNode.TEMPLATE_CONFIGURATION_URL, configUrl);
+            }
+            if (needsCreationScriptUrl) {
+                clonedSpec.configure(MySqlNode.CREATION_SCRIPT_URL, (String) null);
+            }
+            if (needsCreationScriptContents) {
+                clonedSpec.configure(MySqlNode.CREATION_SCRIPT_URL, (String) null);
+            }
+            return clonedSpec;
+        } else {
+            return spec;
+        }
+    }
+
+    private boolean isKeyConfigured(EntitySpec<?> spec, ConfigKey<?> key) {
+        return spec.getConfig().containsKey(key) || spec.getFlags().containsKey(key.getName());
+    }
+
+    @Override
+    protected Entity createNode(Location loc, Map<?, ?> flags) {
+        Entity node = super.createNode(loc, flags);
+        if (!IS_MASTER.apply(node)) {
+            ServiceNotUpLogic.updateNotUpIndicator((EntityLocal)node, MySqlSlave.SLAVE_HEALTHY, "Replication not started");
+
+            FunctionFeed.builder()
+                .entity((EntityLocal)node)
+                .period(Duration.FIVE_SECONDS)
```

Comment:
I don't see how it would be less magical, unless we make it configurable, but not worth it at this point.